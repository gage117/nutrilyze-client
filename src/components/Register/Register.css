/* Changes the App container to flex */
#Register__main-flex {
  min-height: 47rem;
  height: calc(100vh - 5rem);
  display: flex;
  justify-content: center;
  align-items: center;
}

.Register__container {
  padding: 1rem;
  min-width: 25rem;
  background-color: #27496d;
  border-radius: .75rem;
}

.Register__container > h2 {
  margin: 2rem 0rem 3rem;
  font-size: 4rem;
  text-align: center;
  color: whitesmoke;
}

/* ==User Name== */
.Register__user_name-div {
  width: 20rem;
  margin: 1.5rem auto;
}
.Register__user_name-div > label {
  display: block;
  font-size: 2.2rem;
  color: whitesmoke;
  margin-bottom: .5rem;
}
#Register__user_name {
  border-radius: .3rem;
  width: 20rem;
  height: 3rem;
}

/* ==Password== */
.Register__password-div {
  width: 20rem;
  margin: 1.5rem auto;
}
.Register__password-div > label {
  display: block;
  font-size: 2.2rem;
  color: whitesmoke;
  margin-bottom: .5rem;
}
#Register__password {
  border-radius: .3rem;
  width: 20rem;
  height: 3rem;
}

/* ==Name== */
.Register__name-div {
  width: 20rem;
  margin: 1rem auto;
}
.Register__name-div > label {
  display: block;
  font-size: 2.2rem;
  color: whitesmoke;
  margin-bottom: .5rem;
}
#Register__name {
  border-radius: .3rem;
  width: 20rem;
  height: 3rem;
}

/* ==Register Button== */
.Register__submit-button-div {
  padding: .25rem 0;
  margin: 2.5rem 0 1rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
.Register__submit-button {
  display: inline-block;
  cursor: pointer;
  text-decoration: none;
  font-size: 2.3rem;
  padding: .5rem 1.25rem;
  margin: .5rem .25rem;
  border: 1px solid black;
  border-radius: .3rem;
  color: whitesmoke;
  background-color: #0c7b93;
}
.Register__submit-button:hover {
  text-decoration: underline;
  background-color: #00a8cc;
}

.Register__or {
  margin: .5rem 0;
  font-size: 1.7rem;
  color: whitesmoke;
}

/* ==Login Link== */
.Register__login-link {
  display: inline-block;
  font-size: 2rem;
  margin: 0 .25rem;
  color: #40ceb8;
}

.Register__register-link:hover {
  color: whitesmoke;
}

/* ==Error Element== */
.Register__error {
  font-size: 1.5rem;
  font-weight: bold;
  text-align: center;
  color: rgb(255, 166, 0);
}