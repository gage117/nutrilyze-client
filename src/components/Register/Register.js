import React, { Component } from 'react'
import {Link} from 'react-router-dom'
import AuthService from '../../services/auth-api-service'
import bcrypt from 'bcryptjs'
import './Register.css'

export default class Register extends Component {
  static defaultProps = {
    onRegistrationSuccess: () => {}
  }

  state = { error: null }

  handleSubmit = ev => {
    ev.preventDefault()
    const { name, user_name, password } = ev.target
    bcrypt.hash(password.value, 10)
      .then(hashedPassword => {
        const user = {
          name: name.value,
          user_name: user_name.value,
          password: hashedPassword
        }
    
        AuthService.addUser(user)
          .then(() => {
            name.value = ''
            user_name.value = ''
            password.value = ''
            this.props.onRegistrationSuccess()
          })
      })
  }

  componentDidMount() {
    const element = document.getElementById('App');
    element.classList.toggle('Register__App-flex');
  }

  render() {
    const { error } = this.state
    return (
      <section className='Register__container'>
        <h2>Register</h2>
        <form
          className='Register__form'
          onSubmit={this.handleSubmit}
        >
          <div role='alert'>
            {error && <p className='Register__error'>{error}</p>}
          </div>
          <div className='Register__user_name-div'>
            <label htmlFor='Register__user_name'>
              User name 
            </label>
            <input
              name='user_name'
              type='text'
              required
              id='Register__user_name'>
            </input>
          </div>
          <div className='Register__password-div'>
            <label htmlFor='Register__password'>
              Password 
            </label>
            <input
              name='password'
              type='password'
              required
              id='Register__password'>
            </input>
          </div>
          <div className='Register__name-div'>
            <label htmlFor='Register__name'>
              Name
            </label>
            <input
              name='name'
              type='text'
              required
              id='Register__name'>
            </input>
          </div>
          <div className='Register__submit-button-div'>
            <button className='Register__submit-button' type='submit'>
              Register
            </button>
            <p className='Register__or'>or</p>
            <Link className='Register__login-link' to='/login'>login</Link>
          </div>
        </form>
      </section>
    )
  }

  componentWillUnmount() {
    const element = document.getElementById('App');
    element.classList.toggle('Register__App-flex');
  }
}
