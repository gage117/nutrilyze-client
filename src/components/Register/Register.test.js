import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import {BrowserRouter} from 'react-router-dom'
import Register from './Register'

describe('Register', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <BrowserRouter>
        <Register />
      </BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  it('renders the UI as exptected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Register />
        </BrowserRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})