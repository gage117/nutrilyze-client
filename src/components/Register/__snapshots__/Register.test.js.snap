// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register renders the UI as exptected 1`] = `
<main
  id="Register__main-flex"
>
  <section
    className="Register__container"
  >
    <h2>
      Register
    </h2>
    <form
      className="Register__form"
      onSubmit={[Function]}
    >
      <div
        role="alert"
      />
      <div
        className="Register__user_name-div"
      >
        <label
          htmlFor="Register__user_name"
        >
          User name
        </label>
        <input
          id="Register__user_name"
          name="user_name"
          required={true}
          type="text"
        />
      </div>
      <div
        className="Register__password-div"
      >
        <label
          htmlFor="Register__password"
        >
          Password
        </label>
        <input
          id="Register__password"
          name="password"
          required={true}
          type="password"
        />
      </div>
      <div
        className="Register__name-div"
      >
        <label
          htmlFor="Register__name"
        >
          Name
        </label>
        <input
          id="Register__name"
          name="name"
          required={true}
          type="text"
        />
      </div>
      <div
        className="Register__submit-button-div"
      >
        <button
          className="Register__submit-button"
          type="submit"
        >
          Register
        </button>
        <p
          className="Register__or"
        >
          or
        </p>
        <a
          className="Register__login-link"
          href="/login"
          onClick={[Function]}
        >
          login
        </a>
      </div>
    </form>
  </section>
</main>
`;
